src/components/file/File.js

```diff
@@ -744,7 +744,7 @@ export default class FileComponent extends Field {
     const fileWithSameNameUploading = this.filesToSync.filesToUpload
       .some(fileToSync => fileToSync.file?.name === file.name);
 
-    const fileWithSameNameUploaded = this.dataValue
+    const fileWithSameNameUploaded = this.dataValue && this.dataValue
       .some(fileStatus => fileStatus.originalName === file.name);
```

Comment:
I'd prefer to make this check even more secure by checking if this.dataValue is an array. Or you can just use lodash, like this:
_.some(this.dataValue, fileStatus => fileStatus.originalName === file.name);
Lodash does all the checking on the incoming value on its own and doesn't throw an error even if this.dataValue is undefine/string/number etc.

Reply:
Replaced check with lodash.some function